### PR TITLE
DOC: Clarify behavior of ``np.lib.scimath.sqrt`` apropos  -0.0

### PR DESCRIPTION
**Pull request by MuhammadMotawe**
This PR adds:
- Note on the floating-point distinction between 0.0 and -0.0
- Examples for having more control over the output in the presence of a negative 0

Closes:
#20160

### CODE REVIEW

File: numpy/core/code_generators/ufunc_docstrings.py [hunk 1]
```diff
@@ -3827,6 +3827,7 @@ def add_newdoc(place, name, doc):
     --------
     lib.scimath.sqrt
         A version which returns complex numbers when given negative reals.
+        Note: 0.0 and -0.0 are handled differently for complex inputs.
```
Comment (melissawm) [inline numpy/core/code_generators/ufunc_docstrings.py:3830]: Not sure this line is needed, I think just having the example below is enough.

File: numpy/lib/scimath.py [hunk 2]
```diff
@@ -234,6 +234,20 @@ def sqrt(x):
...
+
+    >>> np.emath.sqrt(complex(-4.0, 0.0))
+    2j
+    >>> np.emath.sqrt(complex(-4.0, -0.0))
+    -2j
+
+
+
+
+
```
Comment (melissawm) [inline numpy/lib/scimath.py:250]:
```suggestion

```

Comment (melissawm) [reply numpy/lib/scimath.py:250]: (too many blank lines :)

File: numpy/core/code_generators/ufunc_docstrings.py (hunk 1 above)
Comment (HaoZeke) [reply numpy/core/code_generators/ufunc_docstrings.py:3830]: The example is in the generated docs for `lib.scimath.sqrt` and won't show up on the same page though

**Comment by HaoZeke** [comment]
To fix the issue with the extraneous commit:

```bash
git rebase HEAD~3
# Remove the line with seberg's commit
git rebase main
git push -f
```

**Comment by mattip** [comment]
Thanks @MuhammadMotawe, @seberg

**Pull request merged**